public/js/navigator-js/integration/StateViewMap.js

```diff
@@ -43,33 +45,90 @@ this.navigatorjs.integration = this.navigatorjs.integration || {};
 				state = recipeStates[j];
 
 				if (requestedState.contains(state)) {
-					viewInstance = recipe.getViewInstance();
 
-					if (viewInstance.navigatorBehaviors instanceof Array) {
-						_addViewElementToDOM(recipe);
+					_addViewElementToDOM(recipe);
+					var viewInstance = recipe.getViewInstance();
+					if (Array.isArray(viewInstance.navigatorBehaviors)) {
 						_navigator.add(viewInstance, state);
 					}
+
+				} else {
+					if (recipe.isMounted()) {
+						_removeViewElementFromDOM(recipe);
+					}
 				}
 			}
 		}
 	}
 
+	function _removeViewElementFromDOM(recipe) {
```

Comment:
Yes, removing the element from the DOM will keep the view, the listeners and the element exactly as they are, nothing is garbage collected here.

http://stackoverflow.com/questions/12528049/if-a-dom-element-is-removed-are-its-listeners-also-removed-from-memory

You're right that toggling the `display` property of a DOM element is a faster way to hide an element than removing and adding the child node to the DOM.

https://jsperf.com/removing-an-element-from-the-screen/3

But for me this change is better from an abstraction standpoint.  The styles of the element should be determined by the view itself and it's presence in the document should be handled by the router.   In my opinion, when I run a `querySelector` on the page I should not be able to find all the elements that were hidden by the router.